### PR TITLE
Reply in kind 1

### PR DESCRIPTION
**Pull request by bob2402**
#403 

- reply button in message actions
- handle reply state change
- reply indicator in editor
- generate reply tags

### CODE REVIEW

File: app/UI/message-panel.tsx [hunk 1]
```diff
@@ -44,7 +44,7 @@ export type ViewUserDetail = {
     pubkey: PublicKey;
 };
 
-interface DirectMessagePanelProps {
+interface MessagePanelProps {
```
Comment (BlowaterNostr) [inline app/UI/message-panel.tsx:47]: Good catch

File: app/UI/app_update.tsx [hunk 2]
```diff
@@ -640,6 +650,47 @@ export function generateTags(content: string, getEventByID: func_GetEventByID, c
             }
         }
     }
+    if (reply_to_event_id) {
```
Comment (BlowaterNostr) [inline app/UI/app_update.tsx:653]: Maybe we can simplify this to just `e` reply and `p` tagged pubkey. The current logic is hard to follow.

**Review by BlowaterNostr — CHANGES_REQUESTED** [review]

File: app/UI/app_update.tsx [hunk 3]
```diff
@@ -598,11 +599,33 @@ export async function handle_SendMessage(
         }
         events = events_send;
     } else if (args.navigationModel.activeNav == "Public") {
-        const nostr_event = await prepareNormalNostrEvent(ctx, {
-            content: event.text,
-            kind: NostrKind.TEXT_NOTE,
-            tags: generateTags(event.text, args.getEventByID, args.current_relay.url),
-        });
+        let replyToEvent: NostrEvent | undefined;
+        if (event.reply_to_event_id) {
```
Comment (BlowaterNostr) [inline app/UI/app_update.tsx:603]: here we have UI event and nostr event. Rename `event` to `ui_event` to be more readable.

**Comment by BlowaterNostr** [comment]
Overall is fine. Let's wait until the next piece and merge together.

File: app/UI/message-list.tsx [hunk 4]
```diff
@@ -41,6 +42,7 @@ interface Props {
         relayRecordGetter: RelayRecordGetter;
         getEventByID: func_GetEventByID;
     };
+    onReplyToEventIDChange?: (eventID?: string | NoteID) => void;
```
Comment (BlowaterNostr) [inline app/UI/message-list.tsx:45]: If we can get the `NostrEvent` at thie level, passing an `NostrEvent` might be better. `string | NoteID` might be too small for this interface. Up to you.

File: app/UI/nav.tsx [hunk 5]
```diff
@@ -67,10 +67,8 @@ export class NavBar extends Component<Props, State> {
     styles = {
         container:
             `h-screen w-16 flex flex-col gap-y-4 overflow-y-auto bg-[${PrimaryBackgroundColor}] py-8 items-center`,
-        icons: (active: boolean, fill?: boolean) => (
-            `w-6 h-6 ${fill ? "fill-current" : "stroke-current"} text-[${
-                active ? PrimaryTextColor : SecondaryTextColor
-            }]`
+        icons: (active: boolean) => (
+            `w-6 h-6 stroke-current text-[${active ? PrimaryTextColor : SecondaryTextColor}]`
```
Comment (BlowaterNostr) [inline app/UI/nav.tsx:71]: It's better to make
```
const SecondaryTextColor = `text-[#XXXXXX]`
```
in the colors file so that the caller code here does not need to prefix `text-`

**Review by BlowaterNostr — CHANGES_REQUESTED** [review]

**Review by BlowaterNostr — APPROVED** [review]

File: app/UI/nav.tsx (hunk 5 above)
Comment (bob2402) [reply app/UI/nav.tsx:71]: Refactor colors separately later
